src/java/edu/slu/action/AnnotationAction.java

```diff
@@ -111,21 +113,85 @@ public void send_error(String msg, int code){
         }
     }
     
+    /**
+     * Add the __rerum properties array to a given JSONObject. If array may already exist, in which case you need to update certain values.  See below.
+     * Properties for consideration are:
+     *   APIversion        —1.0.0
+     *   history.prime     —if it has an @id, import from that, else "root"
+     *   history.next      —always [] (or null, perhaps)
+     *   history.previous  —if it has an @id, @id
+     *   releases.previous —if it has an @id, import from that, else null
+     *   releases.next     —always [] (or null, perhaps)
+     *   generatedBy       —set to the @id of the public agent of the API Key.
+     *   createdAt         —"addedTime" timestamp in milliseconds
+     *   isOverwritten, isReleased   —always null
+     * 
+     * @param received A potentially optionless JSONObject.
+     * @return configuredObject The same object that was recieved but with the proper __rerum options
+     */
+    public JSONObject configureRerumOptions(JSONObject received){
+        JSONObject configuredObject = received;
+        JSONObject history = new JSONObject();
+        JSONObject releases = new JSONObject();
+        JSONObject rerumOptions = new JSONObject();
+        String history_prime = "";
+        String history_previous = "";
+        String releases_previous = "";
+        String[] history_and_releases_next = new String[0];
+        rerumOptions.element("APIversion", "1.0.0");
+        rerumOptions.element("createdAt", System.currentTimeMillis());
+        rerumOptions.element("isOverwritten", "");
+        rerumOptions.element("isReleased", "");
+        if(received.containsKey("history")){
```

Comment:
This is an insufficient condition. It is possible that the application just didn't persist the `__rerum` object or worse altered it, so the __rerum should be fetched from the object in the store, if there is an @id already.